compiler/transforms/accfg_config_overlap.py

```diff
@@ -0,0 +1,74 @@
+from xdsl.dialects import builtin
+from xdsl.ir import MLContext, Operation, Use
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+
+from compiler.dialects import accfg
+
+
+class BlockLevelSetupAwaitOverlapPattern(RewritePattern):
+    """
+    Inspects a single block for the following structure:
+
+    ```
+     %token = accfg.launch %state ...
+     // ...
+     %next_state = accfg.setup ... from %state ...
+    ```
+
+    And moves the setup op right behind the launch, iff:
+    - The these two ops are the only two uses of the `%state` SSA value
+    """
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: accfg.SetupOp, rewriter: PatternRewriter, /):
+        # ignore setup without an input state
+        if not op.in_state:
+            return
+        # only apply if there are exactly two uses of the state
```

Comment:
This is to not fiddle with the order of the a launch op that has been setup by a fully deduplicated setup?
Or why is this check here?

Reply:
(basically, if there is more than one use, I don't see a point on figuring out what the "last" move is, and it makes all other checks more complicated)